### ISSUE TITLE
Asynchronous ramping of ordinary variables; ramping logic redesign etc

### ISSUE THREAD

**Issue by olofson** [olofson/audiality2#205]
There are many situations where one has to modulate things in scripts, and end up implementing it by incrementally updating state kept in one or more variables. In many "typical synth modulation" cases, #51 is probably the right solution, but often, that's overkill and overly complicated.

When all you need is a linear ramp, it would be simpler, cleaner, more robust, and probably more efficient to set up a variable (or unit control register!) for asynchronous ramping, and just read that when desired. Besides, it has the bonus that these ramping variables may be controlled asynchronously from message handlers, with no need to interfere with the control flow of the main program.

As it is, VM registers (A2S variables and unit control registers alike) just hold plain values, and all the actual ramping logic is in the units, with the VM just updating as needed. A VM register mapped to a unit control register always holds the target value of any ongoing ramp, and there's (currently) no way of reading the actual control value back, as that's hidden inside the respective units - which is also why there simply is no ramping for ordinary VM registers.

We probably want to integrate this better with the VM and A2S, possibly in conjunction with implementing the related #47 and #199.

One A2S design issue that springs to mind is that we might want to do away with the implicit default of ramping any updated control registers over the duration of the next 'd' or 'td'. It was a neat shorthand syntax when A2S was essentially VM assembly language, but maybe it's time for the language to grow up?

**TL;DR:** Asynchronous ramping support for ordinary A2S variables, and possibly a bit of related language redesign and code refactoring.
